### ISSUE TITLE
[BUG] Commands shouldn't need @BotKube prefix if posted as DM to BotKube user

### ISSUE THREAD

**Issue by PrasadG193** [infracloudio/botkube#115]
**Describe the bug**
It feels kind of weird to add @BotKube prefix in the commands when posting as direct message to BotKube

**Expected behavior**
BotKube should serve commands with or without @BotKube prefix if posted as DM

**Comment by PrasadG193** [comment]
Same behaviour observed on Mattermost integration. Reopening the issue

**Comment by rkennedy-tpl** [comment]
has this actually be solved or do I need to perform additional Slack configuration? it responds to at least `ping` but most other commands appear ignored.

![image](https://user-images.githubusercontent.com/76958947/114476340-0552a980-9bc8-11eb-88a1-81072b22b742.png)


```
INFO[2021-04-12T23:25:53Z] Invalid request. Dumping the response. Request: help 
INFO[2021-04-12T23:26:42Z] Invalid request. Dumping the response. Request: notifier status 
INFO[2021-04-12T23:27:40Z] Invalid request. Dumping the response. Request:  notifier status 
```

I tried with `tag: v0.12.0` and also tried on `-latest`. (I got a friendly warning for doing that too!)

![image](https://user-images.githubusercontent.com/76958947/114476288-f0761600-9bc7-11eb-8e63-0b2bbad9c25d.png)
